diff --git a/lib/codeclib.c b/lib/codeclib.c
index 43259fcf..cce5be18 100644
--- a/lib/codeclib.c
+++ b/lib/codeclib.c
@@ -4349,7 +4349,7 @@ static const char evs_amr_io_compact_cmr[8] = {
 };
 
 
-#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(ifunc)
+#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(IFUNC)
 static void mvr2s_dynlib_wrapper(float *in, const uint16_t len, int16_t *out) {
 	evs_syn_output(in, len, out);
 }
diff --git a/lib/mix_buffer.c b/lib/mix_buffer.c
index 48566caf..cdd53ad9 100644
--- a/lib/mix_buffer.c
+++ b/lib/mix_buffer.c
@@ -53,7 +53,7 @@ static void s16_mix_in_c(void *restrict dst, const void *restrict src, unsigned
 }
 
 
-#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(ifunc)
+#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(IFUNC)
 static mix_in_fn_t *resolve_s16_mix_in(void) {
 	if (rtpe_has_cpu_flag(RTPE_CPU_FLAG_AVX512BW))
 		return s16_mix_in_avx512;
