diff --color -ru rtpengine-mr11.5.1.2/lib/codeclib.c src/rtpengine-mr11.5.1.2/lib/codeclib.c
--- rtpengine-mr11.5.1.2/lib/codeclib.c	2023-09-20 10:54:05.663535541 +0000
+++ src/rtpengine-mr11.5.1.2/lib/codeclib.c	2023-09-18 19:50:33.000000000 +0000
@@ -4349,7 +4349,7 @@
 };


-#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(IFUNC)
+#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(ifunc)
 static void mvr2s_dynlib_wrapper(float *in, const uint16_t len, int16_t *out) {
 	evs_syn_output(in, len, out);
 }
diff --color -ru rtpengine-mr11.5.1.2/lib/mix_buffer.c src/rtpengine-mr11.5.1.2/lib/mix_buffer.c
--- rtpengine-mr11.5.1.2/lib/mix_buffer.c	2023-09-20 10:53:44.931068345 +0000
+++ src/rtpengine-mr11.5.1.2/lib/mix_buffer.c	2023-09-18 19:50:33.000000000 +0000
@@ -53,7 +53,7 @@
 }


-#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(IFUNC)
+#if defined(__x86_64__) && !defined(ASAN_BUILD) && HAS_ATTR(ifunc)
 static mix_in_fn_t *resolve_s16_mix_in(void) {
 	if (rtpe_has_cpu_flag(RTPE_CPU_FLAG_AVX512BW))
 		return s16_mix_in_avx512;
