diff --color -urN nats.c-3.6.1/test/CMakeLists.txt nats.c/test/CMakeLists.txt
--- nats.c-3.6.1/test/CMakeLists.txt	2023-03-07 20:17:31.000000000 +0000
+++ nats.c/test/CMakeLists.txt	2023-09-18 08:18:52.490357976 +0000
@@ -1,17 +1,20 @@
 if(NOT BUILD_TESTING)
   return()
 endif()
+
 if(NOT NATS_BUILD_LIB_STATIC)
-   MESSAGE(FATAL_ERROR
-     "Building tests require static library, or run CMake with -DBUILD_TESTING=OFF")
+  MESSAGE(FATAL_ERROR
+    "Building tests require static library, or run CMake with -DBUILD_TESTING=OFF")
   return()
-endif()  
+endif()
 
 # We need this to build the test program
 include_directories(${PROJECT_SOURCE_DIR}/src)
+
 if(NATS_BUILD_WITH_TLS)
   include_directories(${OPENSSL_INCLUDE_DIR})
 endif(NATS_BUILD_WITH_TLS)
+
 if(NATS_BUILD_STREAMING)
   include_directories(${NATS_PROTOBUF_INCLUDE_DIRS})
   include_directories(${PROJECT_SOURCE_DIR}/src/stan)
@@ -29,20 +32,30 @@
 # Read the file 'list.txt' to get all the test names
 file(STRINGS list.txt listOfTestNames)
 
-# For each test name
-foreach(name ${listOfTestNames})
+# See https://github.com/nats-io/nats.c/issues/674
+file(STRINGS flappers.txt listOfSkipFlappers)
 
-  # Create a test and pass the index (start and end are the same)
-  # to the testsuite executable
-  add_test(NAME Test_${name}
-           WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
-           COMMAND testsuite ${testIndex} ${testIndex})
+# See https://github.com/nats-io/nats.c/issues/672
+file(STRINGS no-sanitize.txt listOfSkipSanitize)
 
-  # Make sure the test passes
-  set_tests_properties(Test_${name} PROPERTIES PASS_REGULAR_EXPRESSION "ALL PASSED")
+# For each test name
+foreach(name ${listOfTestNames})
+  if("${name}" IN_LIST listOfSkipFlappers)
+    # skip, see https://github.com/nats-io/nats.c/issues/674
+  elseif(DEFINED ENV{NATS_SKIP_GH_674} AND "${name}" IN_LIST listOfSkipSanitize)
+    # skip, see https://github.com/nats-io/nats.c/issues/672
+  else()
+    # Create a test and pass the index (start and end are the same)
+    # to the testsuite executable
+    add_test(NAME Test_${name}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+      COMMAND testsuite ${testIndex} ${testIndex})
+
+    # Make sure the test passes
+    set_tests_properties(Test_${name} PROPERTIES PASS_REGULAR_EXPRESSION "ALL PASSED")
+    set_tests_properties(Test_${name} PROPERTIES FAIL_REGULAR_EXPRESSION "WARNING\: ThreadSanitizer\:")
+  endif()
 
   # Bump the test index number
   math(EXPR testIndex "${testIndex}+1")
 endforeach()
-
-
diff --color -urN nats.c-3.6.1/test/flappers.txt nats.c/test/flappers.txt
--- nats.c-3.6.1/test/flappers.txt	1970-01-01 00:00:00.000000000 +0000
+++ nats.c/test/flappers.txt	2023-09-18 08:18:52.490357976 +0000
@@ -0,0 +1 @@
+PendingLimitsDeliveredAndDropped
diff --color -urN nats.c-3.6.1/test/no-sanitize.txt nats.c/test/no-sanitize.txt
--- nats.c-3.6.1/test/no-sanitize.txt	1970-01-01 00:00:00.000000000 +0000
+++ nats.c/test/no-sanitize.txt	2023-09-18 08:18:52.490357976 +0000
@@ -0,0 +1,10 @@
+AuthFailToReconnect
+AutoUnsubNoUnsubOnDestroy
+ClientAutoUnsubAndReconnect
+EventLoop
+JetStreamOrderedConsSrvRestart
+JetStreamOrderedConsWithErrors
+ParseStateReconnectFunctionality
+PendingLimitsDeliveredAndDropped
+ReconnectWithTokenHandler
+SrvVersionAtLeast
